### DISCUSSION TITLE
Adding Other Cowrie JSON

### DISCUSSION THREAD

**Discussion by kalpinus** [telekom-security/tpotce#1090] [Q&A]
Hello,

I changed logstash.conf according to [https://github.com/telekom-security/tpotce/wiki/Reconfigure-logstash.conf](https://github.com/telekom-security/tpotce/wiki/Reconfigure-logstash.conf)  to read new json file from other cowrie honeypot as following:

Cowrie

file {
path => ["/data/cowrie/log/cowrie.json"]
codec => json
type => "Cowrie"
}

Cowrie DC01

file {
path => ["/data/cowrie/log/cowriedc01.json"]
codec => json
type => "Cowrie"
}

Also I already copy cowriedc01.json from other cowrie honeypot to the T-Pot and change the permission 664 and owner to tpot:tpot. After that I run tpot again and I can see logstash now use new configuration but I did not see new dst_ip (as other cowrie's IP) in the elasticsearch, looks like logstash just ignore cowriedc01.json.

So how do I make T-Pot process other cowrie's json log from other honeypot and display them to Kibana ?

Thank you

**Comment by kalpinus** [comment]
hello,

any suggestion for this one ? I want create as much as possible cowrie in other infrastructure without need to install many modules just cowrie...

thank you

**Comment by Ganesh-403** [comment]
@kalpinus 
Your approach is generally correct—Logstash **can** monitor multiple Cowrie JSON files using multiple `file` inputs. If the second file is being ignored, the issue is likely not the `type => "Cowrie"` itself, but rather how Logstash is tracking or reading the file.

A few things I'd check:

### 1. Verify Logstash is actually watching the new file

First, confirm the file exists and is readable by the Logstash process:

```bash
ls -l /data/cowrie/log/cowriedc01.json
```

Then check the Logstash logs for any errors related to that input:

```bash
docker logs logstash
```

Look for messages about the file not being found, permission errors, or JSON parsing failures.

---

### 2. Remember how the `file` input works

By default, the Logstash `file` input behaves like `tail -f`:

- it starts reading **from the end** of a file on first discovery,
- and it remembers its read position in a **sincedb** file. :contentReference[oaicite:0]{index=0}

If `cowriedc01.json` already contained historical events when you copied it, Logstash may consider them "old" and won't index them unless new lines are appended.

For testing, you can temporarily configure:

```ruby
file {
  path => "/data/cowrie/log/cowriedc01.json"
  codec => json
  type => "Cowrie"
  start_position => "beginning"
}
```

If Logstash has already created a `sincedb` entry for that file, `start_position` alone won't help because it only applies the **first time** a file is seen. In that case, you'll need to remove/reset the corresponding `sincedb` entry or use a temporary `sincedb_path` while testing. :contentReference[oaicite:1]{index=1}

---

### 3. Check whether new events are being written

If `cowriedc01.json` is just a static copy from another Cowrie instance, Logstash will import it once (if configured to read from the beginning) and then stop.

If your goal is continuous ingestion from another Cowrie honeypot, you'll need a mechanism that continuously updates the file or ships logs (for example, Filebeat). The Cowrie documentation also describes using Filebeat to forward logs to Logstash, which is generally the preferred approach for multiple sensors. :contentReference[oaicite:2]{index=2}

---

### 4. Verify the events are indexed

If Logstash is reading the file but you don't see the expected `dst_ip`, inspect one of the indexed documents in Kibana's **Discover** view.

It's possible that:

- the events are being indexed,
- but the remote Cowrie logs contain different values or field names than expected,
- or another filter in the pipeline is dropping or modifying them.

---

One question that would help narrow this down:

- Is `cowriedc01.json` a **static file copied once**, or is it **continuously updated** by the remote Cowrie instance?

If it's the latter, I'd recommend using a log shipper such as Filebeat rather than manually copying JSON files, as that's the workflow Cowrie documents for sending events to a central Logstash instance. :contentReference[oaicite:3]{index=3}

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
